### PR TITLE
Refactor/views

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?

https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/rails_refactor
Because the view was large and had a lot of logic
### What Changed?

Added RESTful routes
Moved forms into partials
Moved logic out of view

### CODE REVIEW

File: app/controllers/items_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class ItemsController < ApplicationController
-  def home
-    @item = Item.new
-    @items = Item.all
+  def index
```
Comment (albaer) [inline app/controllers/items_controller.rb:2]: Hooray for REST!

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -9,28 +9,28 @@ def create
     if item_params[:name].blank?
       @item = Item.new
       flash[:error] = "Invalid"
-      render :home and return
+      render :index and return
```
Comment (albaer) [inline app/controllers/items_controller.rb:12]: Nice job only changing one big thing at a time. I'm sure it hurts a little to leave this here :)

File: app/views/categories/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<p>
```
Comment (albaer) [inline app/views/categories/_form.html.erb:1]: Nice! Both moving this to a partial and and giving it the conventional name.

File: app/views/items/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<p>
```
Comment (albaer) [inline app/views/items/_form.html.erb:1]: Same here.

File: app/views/items/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<p> <%= flash[:error] %> </p>
+
+<%= render 'form' %>
```
Comment (albaer) [inline app/views/items/index.html.erb:3]: Much cleaner.

**Comment by albaer** [comment]
Great work! Really nice start on this. :shipit:

**Comment by TylerRockwell** [comment]
Thanks, Annie!

**Pull request merged**